src/main/java/org/jabref/logic/l10n/Localization.java

```diff
@@ -1,41 +1,88 @@
 package org.jabref.logic.l10n;
 
-import java.nio.charset.StandardCharsets;
-import java.util.Locale;
-import java.util.MissingResourceException;
-import java.util.Objects;
-import java.util.Optional;
-import java.util.ResourceBundle;
-
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+/**
+ * Provides handling for messages and menu entries in the preferred language of the user.
+ *
+ * Notes: All messages and menu-entries in JabRef are stored in escaped form like "This_is_a_message". This message
+ * serves as key inside the {@link l10n} properties files that hold the translation for many languages.
+ * When a message is accessed, it needs to be unescaped and possible parameters that can appear in a message need to
+ * be filled with values.
+ *
+ * This implementation loads the appropriate language by importing all keys/values from the correct bundle and stores
+ * them in unescaped form inside a {@link LocalizationBundle} which provides fast access because it caches the key-value
+ * pairs.
+ *
+ * The access to this is given by the functions {@link Localization#lang(String, String...)} and
+ * {@link Localization#menuTitle(String, String...)} that developers should use whenever they use strings for the e.g.
+ * GUI that need to be translatable.
+ */
 public class Localization {
-    public static final String RESOURCE_PREFIX = "l10n/JabRef";
-    public static final String MENU_RESOURCE_PREFIX = "l10n/Menu";
+    static final String RESOURCE_PREFIX = "l10n/JabRef";
+    static final String MENU_RESOURCE_PREFIX = "l10n/Menu";
     public static final String BIBTEX = "BibTeX";
 
     private static final Log LOGGER = LogFactory.getLog(Localization.class);
 
-    private static ResourceBundle messages;
-    private static ResourceBundle menuTitles;
+    private static Locale locale;
+    private static LocalizationBundle localizedMessages;
+    private static LocalizationBundle localizedMenuTitles;
 
     private Localization() {
     }
 
-    public static LocalizationBundle getMessages() {
-        return new LocalizationBundle(messages);
+    /**
+     * Public access to all messages that are not menu-entries
+     * @param key The key of the message in unescaped form like "All fields"
+     * @param params Replacement strings for parameters %0, %1, etc.
+     * @return The message with replaced parameters
+     */
+    public static String lang(String key, String... params) {
+        if (localizedMessages == null) {
+            // I'm logging this because it should never happen
+            LOGGER.error("Messages are not initialized in " + Localization.class);
```

Comment:
Yozu can omit the class part when writing a log message, as the Logger is already instantiated for the current class and therefore knows its name.